### PR TITLE
Fix bad links and test them in the future

### PR DESCRIPTION
**Pull request by tgaff**
# ready for review

This fixes more bad links.

Broken links are remarkably easy to add when you're not copy-pasting from the site (since your page may not be live yet). Therefore this also adds tests using lychee https://github.com/lycheeverse/lychee-action/tree/master

We test in ~two~ one case:

1. On a PR - this tests against the markdown itself, not the site.  It does not build with Jekyll.
2. ~After deploy.  This tests against a real true and live site.~ Ran into tooling troubles, I'll do a separate PR with this.

Screen Shot 2023-10-06 at 4 50 42 PM

Failures look like this:

Screen Shot 2023-10-10 at 10 37 52 AM
I think they're fairly easy to interpret.

### CODE REVIEW

File: technical_guides/chef.md [hunk 1]
```diff
@@ -44,7 +44,7 @@
 More Information / References
 =============================
 
-  * [The Berkshelf Way ~ Jamie Winsor](http://www.getchef.com/blog/chefconf-talks/the-berkshelf-way-jamie-winsor/)
+  * The Berkshelf Way ~ Jamie Winsor, formerly at: http://www.getchef.com/blog/chefconf-talks/the-berkshelf-way-jamie-winsor/
```
Comment (tgaff) [inline technical_guides/chef.md:47]: I found only one link to something that matched and it was on a site so laden in ads that safari struggled to load the next page.  I've decided I'd rather not link to them.

File: technical_guides/css.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Store all colors, font families, font weights, and font sizes in variables.
 
 Use variable-heavy math. Avoid magic numbers whenever possible (and comment when necessary). Store all variables in a separate file so you can see where it's possible to consolidate.
 
-You might consider using [Scut](https://davidtheclark.github.io/scut/) for common utilities and tricks.
+You might consider using [Scut](https://ramseyinhouse.github.io/scut) for common utilities and tricks.
```
Comment (tgaff) [inline technical_guides/css.md:13]: Is this still relevant?  @JeffreyATW thoughts?

Comment (JeffreyATW) [reply technical_guides/css.md:13]: It is not.

File: technical_guides/chef.md (hunk 1 above)
Comment (holetse) [reply technical_guides/chef.md:47]: I'd suggest we drop it. The video seems to not exist on the internet anymore, and the slides are pretty useless without the narration.

Comment (tgaff) [reply technical_guides/chef.md:47]: done

File: technical_guides/css.md (hunk 2 above)
Comment (tgaff) [reply technical_guides/css.md]: dropped this link, left the other scut link - feel free to re-work that guide

**Comment by tgaff** [comment]
Testing a PR pre-merge:

Screen Shot 2023-10-09 at 10 03 54 AM

Testing a merge post-deploy:

Screen Shot 2023-10-09 at 10 17 09 AM

**Comment by tgaff** [comment]
I'm finding some serious issues with this now.  I may end up replacing the backend tool with something else for at least one of the runners.  Please don't merge yet.

**Comment by holetse** [comment]
> I'm finding some serious issues with this now. I may end up replacing the backend tool with something else for at least one of the runners. Please don't merge yet.

Holding off on review while you dig into this. Let us know when it’s ready for review again.

**Comment by tgaff** [comment]
Thanks Ned!

* The PR runner's tool was ignoring root relative links - any link that began with `/`
* The same tool was being used as the site-checker as well, however I realized it was missing a lot in this scenario too and there's no good workaround there.  Will likely replace it with the `muffet` tool, though that one's results are harder to interpret.

I'll probably convert this PR to just the PR checker (scanning md files).  And open another PR for the site-checker.

File: _config.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-description: "our best practices for processes, playbooks, languages and tools"
+description: "Our best practices for processes, playbooks, languages and tools"
```
Comment (tgaff) [inline _config.yml:1]: missed this change on a previous PR

**Comment by tgaff** [comment]
Screen Shot 2023-10-10 at 10 37 52 AM

File: technical_guides/chef.md [hunk 4]
```diff
@@ -44,7 +44,7 @@
 More Information / References
 =============================
 
-  * [The Berkshelf Way ~ Jamie Winsor](http://www.getchef.com/blog/chefconf-talks/the-berkshelf-way-jamie-winsor/)
+  * The Berkshelf Way ~ Jamie Winsor
```
Comment (nyscherm) [inline technical_guides/chef.md:47]: If we don't have a link for this anymore, should we just remove it? I did a quick search and found [this YouTube video](https://www.youtube.com/watch?v=Dq_vGxd-jps) that's a follow-up on that talk, and this follow-up is 9 years old now. I realize that getting rid of outdated stuff is beyond the scope of this PR, but maybe this broken link specifically can go?

File: technical_guides/css.md [hunk 5]
```diff
@@ -124,6 +122,6 @@ Use [Stylelint](https://stylelint.io/) along with [stylelint-config-standard](ht
 
 ## Resources
 - [https://developer.mozilla.org/en-US/docs/Web/CSS](https://developer.mozilla.org/en-US/docs/Web/CSS)
-- [http://caniuse.com](http://caniuse.com)
+- [https://caniuse.com](https://caniuse.com)
```
Comment (nyscherm) [inline technical_guides/css.md:125]: Hooray for https

**Review by nyscherm — COMMENTED** [review]
This looks good to me, just one suggestion. I haven't tried running lychee locally but it looks good from the github action logs.

File: technical_guides/chef.md (hunk 4 above)
Comment (tgaff) [reply technical_guides/chef.md:47]: I removed the link.  Citing it as a reference though is still valid I think.

File: .github/workflows/validate-markdown-links.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  pull_request
+
+jobs:
+  linkChecker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Link Checker
+        uses: lycheeverse/lychee-action@v1.8.0
```
Comment (brienw) [inline .github/workflows/validate-markdown-links.yml:15]: 👍 very nice, haven't seen this one before. definitely adding it to some other projects now

File: technical_guides/chef.md (hunk 4 above)
Comment (brienw) [reply technical_guides/chef.md:47]: agreed

**Review by brienw — APPROVED** [review]
LGTM!

File: .github/workflows/validate-markdown-links.yml (hunk 6 above)
Comment (tgaff) [reply .github/workflows/validate-markdown-links.yml:15]: So this is great for scanning documents locally - but it does have some issues.  That `--base .` is absolutely necessary or it just ignores links that start with `/`.

It also isn't good at crawling all of a live sites links (vs. a folder full of files) so I'm setting up the site scanner using a completely different tool called `muffet`.  PR pending soon...

Comment (tgaff) [reply .github/workflows/validate-markdown-links.yml:15]: Issue no 1265 in the lychee repo

**Pull request merged**